sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs

```diff
@@ -168,5 +177,71 @@ internal static string GetOperationName(Activity activity, ref AzMonList MappedT
 
             return activity.DisplayName;
         }
+
+        private static void AddExceptionTelemetryFromActivityExceptionEvents(Activity activity, ref TagEnumerationState monitorTags, string roleName, string roleInstance, string instrumentationKey, List<TelemetryItem> telemetryItems)
+        {
+            if (activity.Events.Any())
+            {
+                foreach (var evnt in activity.Events)
+                {
+                    if (evnt.Name == SemanticConventions.AttributeExceptionEventName)
+                    {
+                        try
+                        {
+                            var exceptionTelemetryItem = new TelemetryItem("Exception", activity, ref monitorTags, roleName, roleInstance, instrumentationKey);
+                            SetExceptionDataDetailsOnTelemetryItem(evnt.Tags, exceptionTelemetryItem);
+                            telemetryItems.Add(exceptionTelemetryItem);
+                        }
+                        catch (Exception ex)
+                        {
+                            AzureMonitorExporterEventSource.Log.WriteError("FailedToExtractExceptionFromActivityEvent", ex);
+                        }
+                    }
+                }
+            }
+        }
+
+        internal static void SetExceptionDataDetailsOnTelemetryItem(IEnumerable<KeyValuePair<string, object>> activityEventTags, TelemetryItem exceptionTelemetryItem)
+        {
+            string exceptionType = null;
+            string exceptionStackTrace = null;
+            string exceptionMessage = null;
+
+            // TODO: update to use perf improvements in .NET7.0
+            foreach (var tag in activityEventTags)
+            {
+                if (tag.Key == SemanticConventions.AttributeExceptionType)
+                {
+                    exceptionType = tag.Value.ToString();
```

Comment:
a dictionary to cache this string would be nice. else we end up with new strings ...

Reply:
added a todo for this to follow up with.